### ISSUE TITLE
Returned matches for preg_match

### ISSUE THREAD

**Issue by shish** [thecodingmachine/phpstan-safe-rule#40]
PHPStan has some Very Special Magic which allows the matches variable to be usefully-typed, but the magic isn't triggered by \Safe\preg_match

```php
<?php

require_once "vendor/autoload.php";

// Dumped type: array{0?: string, 1?: non-empty-string}  
preg_match("/te(.)ting/", "testing patterns", $matches);
\PHPStan\dumpType($matches);

// Dumped type: array<string>|null                       
\Safe\preg_match("/te(.)ting/", "testing patterns", $matches);
\PHPStan\dumpType($matches);

```
